hangups/client.py

```diff
@@ -109,6 +108,11 @@ def __init__(self, cookies, max_retries=5, retry_backoff_base=2):
         # ActiveClientState enum int value or None:
         self._active_client_state = None
 
+    def __del__(self):
+        """explicit cleanup"""
+        if self._session is not None:
+            self._session.close()
```

Comment:
Why move the session cleanup from `connect` to `__del__`?

Reply:
To allow scheduled calls to run further the end of listening:
- start listening
- process events, e.g. schedule reply messages
- stop listening to new events as it would result in more messages to be scheduled
- await sending

Reply:
Let's leave the client lifecycle changes for a separate PR. 

Using `__del__` might be just fine, but best practice seem to be to offer an (async) context manager and/or an explicit `close` method. If we go this route we may also have to address backwards-compatibility with code that expects `connect` to close when it returns.

Reply:
changed in https://github.com/tdryer/hangups/pull/342/commits/3831cae8e4f133e3a3c6aed2d3570cd5151f8f92